### PR TITLE
REMOVE SPROCKETS + SUPPORT VITE

### PR DESCRIPTION
**Pull request by hapiben**
Changes:

- Removed the Sass gem that depends on Sprockets.
- Used the Vite.js tag to include JS.
- Updated the install generator to include `plug.js` entrypoint and add `Plug` package.
- Packaged the old `Plug` assets into a npm package.
- Removed/tidy ups CSS files.

### CODE REVIEW

File: lib/generators/plug/install_generator.rb [hunk 1]
```diff
@@ -11,5 +11,23 @@ def copy_initializer_file
     def mount_engine
       inject_into_file('config/routes.rb', "\n  mount Plug::Engine => '/plug', as: 'plug'\n\n", before: /^end/)
     end
+
+    def setup_vite_assets
+      copy_file 'plug.vite.js', 'app/frontend/entrypoints/plug.js'
+    end
+
+    def add_plug_package
```
Comment (hapiben) [inline lib/generators/plug/install_generator.rb:19]: This adds a package called `@digitalnz/plug` to the host app during the installation via `rails g plug:install` command.

File: src/plug/stylesheets/base.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Trix editor
+@import 'trix/dist/trix';
+@import 'trix-overrides';
+
+@import 'milligram';
```
Comment (hapiben) [inline src/plug/stylesheets/base.scss:5]: `Milligram` and `Trix` are imported via npm package now instead of the minified version.

Comment (paul-mesnilgrente) [reply src/plug/stylesheets/base.scss:5]: That's much better!

**Review by paul-mesnilgrente — APPROVED** [review]
LGTM

File: lib/generators/plug/install_generator.rb (hunk 1 above)
Comment (chigginsboost) [reply lib/generators/plug/install_generator.rb:19]: Nice!

**Comment by chigginsboost** [comment]
super nit: Do you think you could change the title of this MR to mention that it's for Plug? I feel like we might have trouble finding this MR down the line without it :) otherwise it looks really good. Nice!

**Review by chigginsboost — APPROVED** [review]

**Review by richardmatthewsdev — APPROVED** [review]

**Pull request merged**